tests/src/main.rs

```diff
@@ -58,90 +71,99 @@ fn main() {
     // Pull in rustc_codegen_spirv as a dynamic library in the same way
     // spirv-builder does.
     let codegen_backend_path = find_rustc_codegen_spirv();
-    let libs = build_deps(&deps_target_dir, &codegen_backend_path);
 
-    run_mode(
-        "ui",
+    let runner = Runner {
         opt,
         tests_dir,
         compiletest_build_dir,
-        &deps_target_dir,
-        &codegen_backend_path,
-        &libs,
-    );
+        deps_target_dir,
+        codegen_backend_path,
+    };
+
+    runner.run_mode("ui");
 }
 
 // FIXME(eddyb) a bunch of these functions could be nicer if they were methods.
```

Comment:
At least one function is now a method, feel free to remove this comment or move it further down.